### ISSUE TITLE
[Bug]: Improvements to LC0092 - enum values starting with a digit

### ISSUE THREAD

**Issue by TKapitan** [ALCops/Analyzers#321]
### Analyzer

LinterCop

### Issue type

False positive (diagnostic fires when it shouldn't)

### Rule ID

LC0092

### Description

We see across multiple apps warnings from LC0092 for structures we think should be allowed. Mostly about enum values that start with a number. This is not described in AL guidelines or Microsoft best practices as not allowed/recommended and we believe it should be allowed.

### AL code to reproduce

<img width="318" height="275" alt="Image" src="https://github.com/user-attachments/assets/d52918a3-95d4-4962-b648-21375ddcbd2b" />

**Comment by Arthurvdv** [comment]
I'm unsure if I should change the default behavior of this. While Microsoft's best practices don't explicitly prohibit digit-starting enum values, the convention is inferred from:

1. All Microsoft examples use PascalCase enum values (None, Bronze, Silver, Gold)
2. AL Guidelines [state](https://alguidelines.dev/docs/bestpractices/variable-naming/) that names requiring quotation marks should be avoided: "_characters that would make quotation marks around a variable necessary must be omitted._" Since "6dpmn" requires quotes (digitstart), it falls outside recommended practice.
3. Separation of concerns: In AL, enum values have a separate Caption property for display text. The enum value name is a programmatic identifier (referenced as MyEnum::"6dpmn" in code), while Caption is the user-facing label.

The rule is designed to be configurable as naming conventions are not one-size-fits-all and you should be ableto tailor patterns to your domain.  You can easily override the default behavior by adding this to your alcops.json:

```json
{
  "NamingPatterns": {
    "EnumValue": {
      "AllowPattern": "^[A-Z0-9]",
      "AllowDescription": "should start with an uppercase letter or digit"
    }
  }
}
```

This changes the rule to accept enum values starting with any letter or digit while still flagging names starting with special characters.

For reference, the full documentation on configuring LC0092 is available at: https://alcops.dev/docs/analyzers/lintercop/lc0092/

**Comment by TKapitan** [comment]
I don't agree with your 2. point and the current implementation of AL Cops is also different. Enum value is not a variable but more similar to object name. 

It allows all special characters (except commas) and spaces, so quotation marks are standard practice. None of the special characters trigger warnings (except commas).

<img width="734" height="309" alt="Image" src="https://github.com/user-attachments/assets/deaa827a-122b-4c1f-be88-75f144ef0712" />

PROBLEM 2: I've now added AL Cops to extension from another partner who has mandatory prefix that starts with 1 (1XX) and now all objects and enum extensions values show the same warning. I guess this is not common, but there are partners with prefix starting with a digit, either for PTE or AppSource

**Comment by Arthurvdv** [comment]
Sorry for keeping you waiting on this.

After thinking it through, you've convinced me by the argument that *"an enum value is not a variable, but is more similar to an object name."* Given that perspective, I agree that enum values should not be treated the same way as procedures/variables by this rule.

I'm therefore planning to exclude enum values from the rule by default. For teams that prefer the current behavior, it can still be enabled through the `alcops.json` configuration.

P.S. I believe the mandatory prefix requirement should already be covered by this rule, although that depends on the `AppSourceCop.json` configuration being set up correctly. I'll take another look to verify that this is indeed the case.

**Comment by Arthurvdv** [comment]
Hi @TKapitan,

I've now addressed this in ALCops version `0.9.0-beta.2`.

Thanks again for raising the issue and challenging the wrong assumption behind the rule.

**Comment by Arthurvdv** [comment]
Hi @TKapitan,

The fix for this issue has now been released as part of the stable ALCops `0.9.0` release.

I'm going to close this issue for now. If you're still experiencing problems or notice anything that doesn't behave as expected, please feel free to reopen it.
